File: src/main/java/au/com/anthonybruno/gymbuddy/workout/WorkoutServiceImpl.java
package au.com.anthonybruno.gymbuddy.workout;

import java.util.List;

public class WorkoutServiceImpl implements WorkoutService {

    private final WorkoutRepository workoutRepository = new WorkoutRepository();

    @Override
    public List<Workout> getWorkouts(int userId) {
        return workoutRepository.getWorkouts(userId);
    }

    @Override
    public Workout addWorkout(int userId, Workout workout) {
        return workoutRepository.addWorkout(userId, workout);
    }
}

File: backend/src/main/kotlin/dev/anthonybruno/gymbuddy/workout/WorkoutServiceImpl.kt
package dev.anthonybruno.gymbuddy.workout

class WorkoutServiceImpl : WorkoutService {

    private val workoutRepository = WorkoutRepository()

    override fun getWorkouts(userId: Long): List<Workout> {
        return workoutRepository.getWorkouts(userId)
    }

    override fun addWorkout(userId: Long, workout: Workout): Workout {
        return workoutRepository.addWorkout(userId, workout)
    }
}
